grammarinator/process.py

```diff
@@ -98,10 +98,21 @@ def __init__(self, idx, conditions):
         self.idx = idx
         self.conditions = conditions
 
-    def literal_alternatives(self):
-        if self.out_neighbours and all(len(alt.out_neighbours) == 1 and isinstance(alt.out_neighbours[0], LiteralNode) for alt in self.out_neighbours):
-            return [repr(alt.out_neighbours[0].src) for alt in self.out_neighbours]
-        return None
+    def simple_alternatives(self):
+        # Check if an alternation contains simple alternatives only (single literals or rule references).
+        # It returns a triplet with at least 2 None values. The first element of the triplet is a list
+        # if the alternation contains simple literals only and contains the literal values. The second
+        # element is a list with rule ids if the alternation contains rule references only. The third
+        # element is a list of literal values and rule ids if the alternation contains single literals
+        # and rule references only.
+        if not self.out_neighbours or any(len(alt.out_neighbours) != 1 or not isinstance(alt.out_neighbours[0], (LiteralNode, RuleNode)) for alt in self.out_neighbours):
+            return None, None, None
+
+        if all(isinstance(alt.out_neighbours[0], LiteralNode) for alt in self.out_neighbours):
+            return [repr(alt.out_neighbours[0].src) for alt in self.out_neighbours], None, None
+        if all(isinstance(alt.out_neighbours[0], RuleNode) for alt in self.out_neighbours):
+            return None, [alt.out_neighbours[0].id for alt in self.out_neighbours], None
+        return None, None, [(repr(alt.out_neighbours[0].src), None) if isinstance(alt.out_neighbours[0], LiteralNode) else (None, alt.out_neighbours[0].id) for alt in self.out_neighbours]
```

Comment:
`repr` should not get called on literals, neither at this line nor at some lines above. The result of this function will be used in jinja template(s) which may eventually support multiple languages, not only Python. And `repr` may add single or double quotes around `src`, which may not fit well with other programming languages. So, `repr` should happen on the Python template side. (And if and when another language gets supported, it should convert `src` to its language literal format as it sees fit.)